zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+
+
+# Detect the platform reliably
+if(ZEPHYR_BASE)
+  if (NOT CONFIG_ZXA)
+    return()
+  endif()
+  set(ZEPHYR YES)
+else()
+  message(FATAL_ERROR "ZEPHYR_BASE needs to be defined for Zephyr builds")
+endif()
+
+#Do not allow in source builds
+set(CMAKE_DISABLE_SOURCE_CHANGES ON)
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
+
+get_filename_component(MODULE_NAME ${ZEPHYR_CURRENT_MODULE_DIR} NAME)
+add_definitions(
+  -DMODULE_NAME="${MODULE_NAME}"
+  )
```

Comment:
Should probably get updated when we have ZEPHYR_CURRENT_MODULE_NAME variables around.

Reply:
Isn't it more important for the impatient to first find out if they even care about this module (what it is, and what it isn't)?

Reply:
I agree that the README should also include cut-and-paste examples for Windows (and all development environments). It also should be presented in the same format, style, and location as in `zephyr` documentation (e.g. .rst tabs). That can be done as an enhancement.

For this PR we need to focus on structure and patterns/precedence so we have repository content that can be used to set up CI and release processing.

Reply:
> Isn't it more important for the impatient to first find out if they even care about this module (what it is, and what it isn't)?
> @mbolivar-nordic Do you have any ideas for making it simpler for the impatient?

Even Markdown supports forward references to later on in the document, right?

So this could be done with sentence or two that says `This is an example application module for Zephyr` that links to somewhere else, followed immediately by a quickstart section that people can copy/paste.

Reply:
>I agree that the README should also include cut-and-paste examples for Windows (and all development environments). It also should be presented in the same format, style, and location as in zephyr documentation (e.g. .rst tabs). That can be done as an enhancement. 

Not sure I follow. You can always do things like:

```
# Windows:
something-for-cmd.exe

# macOS and Linux:
something-for-bash
```

In a  code block without having to wait for sphinx tabs.